### PR TITLE
feat: adds voting tx locale/icon

### LINKED ISSUE #2445: [Task]: Add missing voting tx history locale, icon & color
### Task description

It should say vote/unvote, use the sidebar voting icon, and lets leave the color the same

image

### Requirements

TBD

### Acceptance criteria

TBD

### Creation checklist

- [X] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by jeeanribeiro**
## Summary
Adds logic for showing participation events tx icon and locale for each ParticipationAction

## Relevant Issues
closes #2445 

## Type of Change
Please select any type below that is applicable to your changes, and delete those that are not.
- [ ] __Breaking__ - any change that would cause existing functionality to not work as expected
- [ ] __Chore__ - refactoring, build scripts or anything else that isn't user-facing
- [ ] __Docs__ - changes to the documentation
- [ ] __Fix__ - a change which fixes an issue
- [x] __New__ - a change which implements a new feature
- [ ] __Update__ - a change which updates existing functionality

### Platforms
Please select any platforms where your changes have been tested.
- __Desktop__
	- [ ] MacOS
	- [x] Linux
	- [ ] Windows
- __Mobile__
	- [ ] iOS
	- [ ] Android

## Checklist
Please tick all of the following boxes that are relevant to your changes, and delete those that are not.
- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/ActivityRow.svelte [hunk 1]
```diff
@@ -43,27 +52,32 @@
...
-
-        return `${
-            !txPayload.data.essence.data.incoming && !isParticipationPayload(txPayload) ? '-' : ''
-        }${formatUnitBestMatch(txPayload.data.essence.data.value, true, 2)}`
-    }
-
     $: senderAddress = sendAddressFromTransactionPayload(payload)
     $: receiverAddresses = receiverAddressesFromTransactionPayload(payload)
+    $: participationAction = getMessageParticipationAction(id)
+    $: {
```
Comment (Tuditi) [inline packages/shared/components/ActivityRow.svelte:62]: Maybe this should be a function?

**Review by Tuditi — CHANGES_REQUESTED** [review]
Getting in shape, there are some minor thing that need to be changed. I also have some issues with icons if the ID's are wrong. This has to be tackled with a stored array of past event ID's or an even better solution.

File: packages/shared/components/ActivityRow.svelte [hunk 2]
```diff
@@ -89,7 +99,7 @@
                 if (includeFullSender) {
                     accountAlias = acc.alias
                 }
-                initialsColor = getColor($activeProfile, acc.id)
+                initialsColor = getColor($activeProfile, acc.id) as string
```
[the remark this answers is not in the document]
Comment (begonaalvarezd) [reply packages/shared/components/ActivityRow.svelte:102]: This line has been removed in `develop` anyway, now activity row doesnt depend on wallet color anymore

File: packages/shared/lib/participation/participation.ts [hunk 3]
```diff
@@ -94,3 +104,29 @@ export const getAccountParticipationAbility = (account: WalletAccount): AccountP
...
+ */
+export const getMessageParticipationAction = (messageId: string): ParticipationAction => {
+    const matchedHistoryItem = get(participationHistory)?.find((item) => item.messageId === messageId)
+    if (matchedHistoryItem?.action) {
+        return matchedHistoryItem.action
+    } else {
+        const stakingEndMilestoneIndexes = get(participationEvents)
+            ?.filter((event) => event.eventId === ASSEMBLY_EVENT_ID || event.eventId === SHIMMER_EVENT_ID)
+            ?.map((event) => event.information?.milestoneIndexEnd)
+        const messageMilestone = 0
```
[the remark this answers is not in the document]
Comment (jeeanribeiro) [reply packages/shared/lib/participation/participation.ts:125]: Sorry, I've not added any comments and also forgot about it.
The idea is to have the latest milestone before the treasury event commencing state, so by default if a participate tx is before that milestone it should show as staking.
So we don't have this milestone yet, I will include a TODO comment

**Review by Tuditi — APPROVED** [review]
LGTM!

**Pull request merged**